File: Java/Arrays/problem_1099. Two Sum Less Than K.java
import java.util.Arrays;

class Solution {
    public int twoSumLessThanK(int[] nums, int k) {
        Arrays.sort(nums);
        int pointer1 = 0;
        int pointer2 = nums.length - 1;
        int ans = -1;
        while (pointer1 < pointer2) {
            int sum = nums[pointer1] + nums[pointer2];
            if (sum < k) {
                if (sum > ans) {
                    ans = sum;
                }
                pointer1++;
            } else {
                pointer2--;
            }
        }
        return ans;

    }
}

File: Python/Arrays/problem_1099. Two Sum Less Than K.py
class Solution:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        req = -1
        for i in range(0, len(nums)):
            for j in range(i+1, len(nums)):
                sum_of_two = nums[i] + nums[j]
                if(sum_of_two > req and sum_of_two < k):
                    req = sum_of_two
        return req
